### PR TITLE
fix: warn on invalid cookie name characters

### PR DESCRIPTION
**Pull request by eltigerchino**
closes https://github.com/sveltejs/kit/issues/12799

Adds a warning at dev time of which characters in the cookie name are invalid according to the RFC

---

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This message body should clearly illustrate what problems it solves.
- [ ] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [ ] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [ ] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpm changeset` and following the prompts. Changesets that add features should be `minor` and those that fix bugs should be `patch`. Please prefix changeset messages with `feat:`, `fix:`, or `chore:`.

### Edits

- [x] Please ensure that 'Allow edits from maintainers' is checked. PRs without this option may be closed.

### CODE REVIEW

File: .changeset/old-points-tell.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@sveltejs/kit': patch
+---
+
+chore: warn on invalid cookie name characters
```
Comment (eltigerchino) [inline .changeset/old-points-tell.md:5]:
```suggestion
fix: warn on invalid cookie name characters
```

File: packages/kit/src/runtime/server/cookie.js [hunk 2]
```diff
@@ -113,6 +117,16 @@ export function get_cookies(request, url, trailing_slash) {
 		 * @param {import('./page/types.js').Cookie['options']} options
 		 */
 		set(name, value, options) {
+			// TODO: remove this check in 3.0
+			if (DEV) {
```
Comment (benmccann) [inline packages/kit/src/runtime/server/cookie.js:121]: I think I might show this warning in prod too. It's possible that users have different cookie names in dev and prod and we'd still want them to be aware if it's only happening in prod

Comment (eltigerchino) [reply packages/kit/src/runtime/server/cookie.js]: I've removed the DEV check

File: packages/kit/src/runtime/server/cookie.js [hunk 3]
```diff
@@ -113,6 +116,14 @@ export function get_cookies(request, url, trailing_slash) {
 		 * @param {import('./page/types.js').Cookie['options']} options
 		 */
 		set(name, value, options) {
+			// TODO: remove this check in 3.0
+			const illegal_characters = name.match(INVALID_COOKIE_CHARACTER_REGEX);
+			if (illegal_characters) {
```
Comment (dummdidumm) [inline packages/kit/src/runtime/server/cookie.js:121]: Should we only warn once per cookie name? Else you might get flooded depending on how often you use them

Comment (benmccann) [reply packages/kit/src/runtime/server/cookie.js:121]: I dunno. We really don't want folks to miss this or their apps will break. If it's just one line in a thousand that their app logs then they might overlook it. I don't think this is terrible. Most people won't be affected, but we really do want folks to see it when they are

**Pull request merged**